crates/runtime/src/component/blobstore.rs

```diff
@@ -77,42 +84,72 @@ impl container::HostContainer for Ctx {
             .table
             .get(&container)
             .context("failed to get container")?;
-        match self.handler.get_data(container, name, start..=end).await {
-            Ok(stream) => {
+        match wrpc::wrpc::blobstore::blobstore::get_container_data(
+            &self.handler,
+            Some(ReplacedInstanceTarget::BlobstoreContainer),
+            &wrpc::wasi::blobstore::types::ObjectId {
+                container: container.to_string(),
+                object: name,
+            },
+            start,
+            end,
+        )
+        .await?
+        {
+            (Ok(stream), io) => {
+                if let Some(io) = io {
+                    // TODO: Move this into the runtime
```

Comment:
I think this TODO can be removed

Reply:
I don't think so, what should happen here is that we should store the `io` alongside the `stream` in the resource table and poll on read. Currently there's no streaming - by fully awaiting the `io` we fully receive the data and only then store it in the resource table

Reply:
https://github.com/wasmCloud/wasmCloud/issues/2475